cv-data-controller/src/main/java/com/trihydro/cvdatacontroller/controller/ActiveTimHoldingController.java

```diff
@@ -206,22 +168,55 @@ public ResponseEntity<List<ActiveTimHolding>> getActiveTimHoldingForRsu(@PathVar
         } catch (SQLException e) {
-            e.printStackTrace();
+            log.error("Error getting active tim holding for rsu", e);
             return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(holdings);
-        } finally {
-            try {
-                // close prepared statement
-                if (statement != null)
-                    statement.close();
-                // return connection back to pool
-                if (connection != null)
-                    connection.close();
-                // close result set
-                if (rs != null)
-                    rs.close();
-            } catch (SQLException e) {
-                e.printStackTrace();
+        }
+    }
+
+    @RequestMapping(value = "/get-all", produces = "application/json", method = RequestMethod.GET)
+    public ResponseEntity<List<ActiveTimHolding>> getAllRecords() {
+        List<ActiveTimHolding> holdings = new ArrayList<>();
+
+        String query = "select * from active_tim_holding";
+
+        try (Connection connection = dbInteractions.getConnectionPool(); Statement statement = connection.createStatement(); ResultSet rs = statement.executeQuery(query)) {
+            // convert to ActiveTim object
+            while (rs.next()) {
+                ActiveTimHolding activeTimHolding = new ActiveTimHolding();
+                activeTimHolding.setActiveTimHoldingId(rs.getLong("ACTIVE_TIM_HOLDING_ID"));
+                activeTimHolding.setClientId(rs.getString("CLIENT_ID"));
+                activeTimHolding.setDirection(rs.getString("DIRECTION"));
+                activeTimHolding.setRsuTargetId(rs.getString("RSU_TARGET"));
+                activeTimHolding.setSatRecordId(rs.getString("SAT_RECORD_ID"));
+                activeTimHolding.setStartPoint(new Coordinate(rs.getBigDecimal("START_LATITUDE"), rs.getBigDecimal("START_LONGITUDE")));
+                activeTimHolding.setEndPoint(new Coordinate(rs.getBigDecimal("END_LATITUDE"), rs.getBigDecimal("END_LONGITUDE")));
+                activeTimHolding.setDateCreated(rs.getString("DATE_CREATED"));
+                int rsu_index = rs.getInt("RSU_INDEX");
+                if (!rs.wasNull()) {
+                    activeTimHolding.setRsuIndex(rsu_index);
+                }
+                activeTimHolding.setExpirationDateTime(rs.getString("EXPIRATION_DATE"));
+                activeTimHolding.setPacketId(rs.getString("PACKET_ID"));
+                holdings.add(activeTimHolding);
             }
+            return ResponseEntity.ok(holdings);
+        } catch (SQLException e) {
+            log.error("Error getting all active tim holdings", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(holdings);
+        }
+    }
+
+    @RequestMapping(value = "/delete/{activeTimHoldingId}", produces = "application/json", method = RequestMethod.DELETE)
+    public ResponseEntity<Boolean> deleteActiveTimHolding(@PathVariable Long activeTimHoldingId) {
```

Comment:
suggestion(non-blocking): this may be a good option to pass in a list instead, and allow deleting multiple.

Reply:
I agree, this has been addressed!